fossas/spectrometer
# IAT: Register user defined binary

# Overview

Support registering user-defined IAT binaries as described in [the IAT RFC](http://go/iat-design-doc).

## Acceptance criteria

We can register the fingerprints of the contents of a directory in the FOSSA server using a set of data provided the user that mirrors the data provided for user-defined custom dependencies.

## Testing plan

1. Run a local instance of `fossa`.
2. Run a command to register binary fingerprints as a user-defined dependency. Example:

```shell
cabal run fossa -- experimental-vsi-register-binary-custom-dependency -e 'http://localhost:9578' --fossa-api-key <key> --name <name> --version <version> --license <license>  <dir containing binaries>
```

3. Verify that the binaries appear in the database.

## Risks

Since this is a totally new subcommand I think this is not very risky.

## Checklist


I don't think that tests are super useful for this; everything that I'd normally test is compiler enforced via types.
